.ci/settings.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+
+printf "%s\n" "<settings>
+    <servers>
+        <server>
+            <id>sonatype-nexus-snapshots</id>
+            <username>romanivanov</username>
+            <password>password</password>
+        </server>
+        <server>
+            <id>sonatype-nexus-staging</id>
+            <username>romanivanov</username>
+            <password>password</password>
+        </server>
+        <server>
+            <id>sourceforge-new-site</id>
+            <username>romanivanov</username>
+            <password>password</password>
+        </server>
+    </servers>
+
+    <profiles>
+      <profile>
+          <!-- "mvn -e --no-transfer-progress release:prepare .... -Pgpg"
+               and it is possible to skip " -Dgpg.passphrase=xxxx" at "-Darguments"
+               # shellcheck disable=SC1001
+               Use "gpg '\-\-'list-keys", "pub   1024D/C6EED57A 2010-01-13"
+                    to get "gpg.keyname" in example it is  value "C6EED57A"
+          -->
+          <id>gpg</id>
+          <properties>
+              <gpg.passphrase>X</gpg.passphrase>
+              <gpg.keyname>X</gpg.keyname>
+          </properties>
+      </profile>
+    </profiles>
+</settings>" > ~/.m2/settings.xml
```

Comment:
Please do backup of target file if it is different from what you generate.

I think it will be bet to have settings.xml save as file, and she'll script just do copy to target as search/replace of secrets from env variable to file

Reply:
create file release-settings.xml at https://github.com/checkstyle/checkstyle/tree/master/.ci
create prepare-settings.sh file that copy release-settings.xml to .ci-temp folder, takes from environment variables:
SONATYPE_USER
SONATYPE_PWD
GPG_PASSPHRASE
GPG_KEY
.....
and put values of them in generated file in .ci-temp
after that check if generated file is different in ~/.m2/settigns.xml if yes - macke backup of original file to ~/.m2/settigns.xml.backup.{date}
copy generared file to ~/.m2/settigns.xml

Reply:
Please check PR description for the tasks done.